src/components/HyloTipTapEditor/extensions/TopicMentions.js

```diff
@@ -0,0 +1,70 @@
+import Mention from '@tiptap/extension-mention'
+import { PluginKey } from 'prosemirror-state'
+import { uniqBy } from 'lodash/fp'
+import asyncDebounce from 'util/asyncDebounce'
+import suggestions from './suggestions'
+import findTopics from 'store/actions/findTopics'
+
+export const TopicMentions = ({ dispatch, groupIds, maxSuggestions, onSelection }) =>
+  Mention
+    .extend({
+      name: 'topic',
+      addStorage () {
+        return {
+          loading: false,
+          groupIds,
+          onSelection
+        }
+      },
+      onUpdate ({ transaction }) {
+        if (this.storage.onSelection) {
+          // Look into `doc.descendents` for possible better or more idiomatic way to get this last node
+          const firstTransactionStepName = transaction?.steps[0]?.slice?.content?.content[0]?.type?.name
+
+          if (firstTransactionStepName && firstTransactionStepName === 'topic') {
+            const attrs = transaction?.steps[0]?.slice?.content?.content[0]?.attrs
+            this.storage.onSelection({ name: attrs.id, id: attrs.id })
+          }
+        }
+      }
+    })
+    .configure({
+      HTMLAttributes: {
+        class: 'topic'
+      },
+      renderLabel: ({ node }) => {
+        return node.attrs.label
+      },
+      suggestion: {
+        char: '#',
+        pluginKey: new PluginKey('topicSuggestion'),
+        render: suggestions.render,
+        items: asyncDebounce(200, async ({ query, editor }) => {
+          // Note: Will show "No Result" while loading results.
```

Comment:
certainly would have a spinner in the ideal situation. but flag for later

Reply:
I may still get to this, but it's pretty far down the list.